include/zenoh-pico/api/encoding.h

```diff
@@ -0,0 +1,408 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+
+#ifndef ZENOH_PICO_API_ENCODING_H
+#define ZENOH_PICO_API_ENCODING_H
+
+#include "zenoh-pico/api/types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+/**
+ * Default encoding values used by Zenoh.
+ *
+ * An encoding has a similar role to Content-type in HTTP: it indicates, when present, how data should be interpreted by
+ * the application.
+ *
+ * Please note the Zenoh protocol does not impose any encoding value nor it operates on it.
+ * It can be seen as some optional metadata that is carried over by Zenoh in such a way the application may perform
+ * different operations depending on the encoding value.
+ *
+ * A set of associated constants are provided to cover the most common encodings for user convenience.
+ * This is particularly useful in helping Zenoh to perform additional wire-level optimizations.
+ *
+ * Register your encoding metadata from a string with :c:func:`z_encoding_from_str`. To get the optimization, you need
+ * Z_FEATURE_ENCODING_VALUES to 1 and your string should follow the format: "<constant>;<optional additional data>"
+ *
+ * E.g: "text/plain;utf8"
+ *
+ * Or you can set the value to the constants directly with this list of constants:
+ */
+
+#if Z_FEATURE_ENCODING_VALUES == 1
+// - Below are Primitives types, supported in all Zenoh bindings
+// Just some bytes.
+//
+// Constant alias for string: `"zenoh/bytes"`.
+//
+// Usually used for types: `uint8_t[]`.
+extern const z_owned_encoding_t ENCODING_ZENOH_BYTES;
```

Comment:
please see how encoding constants are implemented in zenoh-c: https://github.com/eclipse-zenoh/zenoh-c/blob/610a4b9057a703ecba5f098e084bdc40e4e786f3/include/zenoh_commons.h#L2254